infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -179,6 +186,35 @@ public ServerContext(Map<String, Object> serverConfig) {
         nodeLocator = NodeLocator
                 .parseString(serverConfig.get("--address") + ":" + serverConfig.get("<port>"));
         localEndpoint = nodeLocator.toEndpointUrl();
+        streamNames = loadStreamIdToNameMap((String) serverConfig.get("--stream-names-file"));
+    }
+
+    public static Map<UUID, String> loadStreamIdToNameMap(String streamNameFile) {
+        Map<UUID, String> streamNames = new HashMap<>();
+        if (streamNameFile == null) {
+            return streamNames;
+        }
+        try {
```

Comment:
Resource leaks, FileReader and BufferREader won't be closed at all

Reply:
done. Thanks!